llvm/include/llvm/Analysis/TargetTransformInfoImpl.h

```diff
@@ -687,6 +691,15 @@ class TargetTransformInfoImplBase {
     return 1;
   }
 
+  InstructionCost getStridedMemoryOpCost(unsigned Opcode, Type *DataTy,
+                                         const Value *Ptr, bool VariableMask,
+                                         Align Alignment,
+                                         TTI::TargetCostKind CostKind,
+                                         const Instruction *I = nullptr) const {
+    return CostKind == TTI::TCK_RecipThroughput ? TTI::TCC_Expensive
```

Comment:
InstructionCost::getInvalid() is likely a better default here.

For codesize, you could reasonable use TCC_Basic for a *legal* strided load or store.

Reply:
I thought about this. Unfortunately, TTI tends to fallback to the default TTI functions in many cases and just returns 1 in many cases for all kinds of costs. If it is ok to override this default behavior here, I will do it.